### DISCUSSION TITLE
Started to get "error: unexpected argument '--cache-key’ found" on loading mise from zshrc

### DISCUSSION THREAD

**Discussion by andrewcrook** [jdx/mise#3559] [Troubleshooting and bug reports] [answered]
on shell start up have been getting the following error 

```sh
error: unexpected argument '--cache-key' found

  tip: to pass '--cache-key' as a value, use '-- --cache-key'

Usage: usage generate completion <SHELL|BIN|--usage-cmd <USAGE_CMD>>

For more information, try '--help'.
```

I believe I have tracked it down to calling `/Users/XXXX/.local/bin/mise activate zsh` in my `.zshrc`
However, I cannot see what’s using `--cache-key`  in the generating code.

A search in the code base I can only find the following occurrence which is in [completion.rs](https://github.com/jdx/mise/blob/6fa612d51679176ce7832b4ead4e816e774f3961/src/cli/completion.rs#L64)

Any idea how to fix this issue?

**Comment by jdx** [comment]
what is `usage -v`?

**Reply by andrewcrook** [reply]
Hi thanks for answering I have just found the issue the tool `usage` needed updating.

**Comment by andrewcrook** [comment, marked as answer]
I have solved the issue found it [here ](https://github.com/jdx/mise/issues/2874)

```
mise use usage@latest
```
